### PR TITLE
zoekt: handle partially indexed RepositoryRevisions

### PR DESCRIPTION
**Pull request by tsenart**
This commit fixes `IndexedRepoRevs.add` to correctly filter out unindexed `search.RevisionSpecifiers` from a `*search.RepositoryRevisions` with *some* indexed rev specs.

Benchmarks show no real diff:

```
name              old time/op    new time/op    delta
SearchResults-12     1.33s ± 7%     1.29s ± 2%   ~     (p=0.114 n=9+7)

name              old alloc/op   new alloc/op   delta
SearchResults-12     469MB ± 0%     469MB ± 0%   ~     (p=0.955 n=7+8)

name              old allocs/op  new allocs/op  delta
SearchResults-12     5.52M ± 0%     5.52M ± 0%   ~     (p=0.650 n=9+9)
```

### CODE REVIEW

File: internal/search/zoekt/indexed_search.go [hunk 1]
```diff
@@ -84,9 +84,11 @@ func (rb *IndexedRepoRevs) add(reporev *search.RepositoryRevisions, repo *zoekt.
 	// Assume for large searches they will mostly involve indexed
 	// revisions, so just allocate that.
 	var unindexed []search.RevisionSpecifier
-	indexed := make([]search.RevisionSpecifier, 0, len(reporev.Revs))
 
 	branches := make([]string, 0, len(reporev.Revs))
+	reporev = reporev.Copy()
+	indexed := reporev.Revs[:0]
```
Comment (keegancsmith) [inline internal/search/zoekt/indexed_search.go:90]: why is this change needed? The older version which just allocated the correctly sized indexed, isn't that correct?

**Review by keegancsmith — APPROVED** [review]

File: internal/search/zoekt/indexed_search.go (hunk 1 above)
Comment (tsenart) [reply internal/search/zoekt/indexed_search.go:90]: Yes, it's correct, but since we need to instantiate a new `*search.RepositoryRevisions`, we can instead just copy the existing one and do in-place filtering. Both approaches yield the same result.
